common/src/main/java/com/viaversion/viarewind/protocol/v1_8to1_7_6_10/data/VirtualHologramEntity.java

```diff
@@ -38,288 +39,313 @@
+            if (remount) {
+                PacketWrapper detach = PacketWrapper.create(ClientboundPackets1_7_2_5.SET_ENTITY_LINK, user);
+                detach.write(Types.INT, entityIds[1]);
+                detach.write(Types.INT, -1);
+                detach.write(Types.BOOLEAN, false);
+                detach.send(Protocol1_8To1_7_6_10.class);
+            }
+
+            // Don't ask me where this offset is coming from
+            teleportEntity(entityIds[0], locX, (locY + (marker ? 54.3625 : small ? 56 : 57) - 0.16), locZ, 0, 0); // Skull
+
+            if (remount) {
+                teleportEntity(entityIds[1], locX, locY + 56.75, locZ, 0, 0); // Horse
+
+                PacketWrapper attach = PacketWrapper.create(ClientboundPackets1_7_2_5.SET_ENTITY_LINK, null, user);
+                attach.write(Types.INT, entityIds[1]);
+                attach.write(Types.INT, entityIds[0]);
+                attach.write(Types.BOOLEAN, false);
+                attach.send(Protocol1_8To1_7_6_10.class);
+            }
+        }
+    }
+
+    protected void teleportEntity(final int entityId, final double x, final double y, final double z, final float yaw, final float pitch) {
+        final PacketWrapper entityTeleport = PacketWrapper.create(ClientboundPackets1_7_2_5.TELEPORT_ENTITY, user);
+
+        entityTeleport.write(Types.INT, entityId); // entity id
+        entityTeleport.write(Types.INT, (int) (x * 32.0)); // x
+        entityTeleport.write(Types.INT, (int) (y * 32.0)); // y
+        entityTeleport.write(Types.INT, (int) (z * 32.0)); // z
+        entityTeleport.write(Types.BYTE, (byte) ((yaw / 360f) * 256)); // yaw
+        entityTeleport.write(Types.BYTE, (byte) ((pitch / 360f) * 256)); // pitch
+
+        entityTeleport.send(Protocol1_8To1_7_6_10.class);
+    }
+
+    protected void spawnEntity(final int entityId, final int type, final double locX, final double locY, final double locZ) {
+        final PacketWrapper addMob = PacketWrapper.create(ClientboundPackets1_7_2_5.ADD_MOB, user);
+
+        addMob.write(Types.VAR_INT, entityId); // Entity id
+        addMob.write(Types.UNSIGNED_BYTE, (short) type); // Entity type
+        addMob.write(Types.INT, (int) (locX * 32.0)); // X
+        addMob.write(Types.INT, (int) (locY * 32.0)); // Y
+        addMob.write(Types.INT, (int) (locZ * 32.0)); // Z
+        addMob.write(Types.BYTE, (byte) 0); // Yaw
+        addMob.write(Types.BYTE, (byte) 0); // Pitch
+        addMob.write(Types.BYTE, (byte) 0); // Head pitch
+        addMob.write(Types.SHORT, (short) 0); // Velocity x
+        addMob.write(Types.SHORT, (short) 0); // Velocity y
+        addMob.write(Types.SHORT, (short) 0); // Velocity z
+        addMob.write(Types1_7_6_10.ENTITY_DATA_LIST, new ArrayList<>()); // Entity data
+
+        addMob.send(Protocol1_8To1_7_6_10.class);
+    }
+
+    protected void spawnEntitySquid(final int entityId, final int type, final double locX, final double locY, final double locZ) {
```

Comment:
Probably worth just reusing spawnEntity and adding a new parameter for the entity data